File: petclinic-java-gradle-react-tailwind-ts/src/main/java/dev/pollito/petclinic_java_gradle_react_tailwind_ts/pet/PetService.java
package dev.pollito.petclinic_java_gradle_react_tailwind_ts.pet;

import dev.pollito.petclinic_java_gradle_react_tailwind_ts.generated.model.PetDTO;
import java.util.List;
import java.util.Map;

public interface PetService {

    List<PetDTO> findAll();

    PetDTO get(Integer id);

    Integer create(PetDTO petDTO);

    void update(Integer id, PetDTO petDTO);

    void delete(Integer id);

    Map<Integer, Integer> getPetValues();

}

File: petclinic-kt-gradle-thymeleaf-tailwind-webpack-htmx-ts/src/main/kotlin/dev/pollito/petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts/pet/PetService.kt
package dev.pollito.petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts.pet

interface PetService {

    fun findAll(): List<Pet>

    fun `get`(id: Int): Pet

    fun create(pet: Pet): Int

    fun update(id: Int, pet: Pet)

    fun delete(id: Int)

    fun getPetValues(): Map<Int, Int>
}
